Proper avatar implementation

YeahImRose opened the pull request:
I really screwed up the submodules on my first PR, this one has edits and proper submodule stuff

YeahImRose committed:
Avatar implementation

YeahImRose committed:
Avatar implementation

YeahImRose committed:
Wrong location

YeahImRose committed:
Move and tweak avatar class

YeahImRose committed:
Proper avatar PR

RemieRichards commented on osu.Game/Overlays/UserButton.cs:
```diff
@@ -0,0 +1,150 @@
+    {
+        public const float WIDTH = 60;
+
+        public Sprite Image
```
it's good to group these with the thing they expose.
eg: this would be just below the definition for ```DrawableAvatar```

RemieRichards commented on osu.Game/Overlays/UserButton.cs:
```diff
@@ -0,0 +1,150 @@
+        protected SpriteText DrawableText;
+        protected Box HoverBackground;
+        private FlowContainer tooltipContainer;
+        private SpriteText tooltip1;
```
these names aren't very descriptive, the Main/Sub distinction you used in the public properties would be fine if you reused it:

```
private tooltipSub;
public TooltipSub;
```

YeahImRose committed:
Fix accidental OsuLogo edits

Lunar-Dust commented on osu.Game/Overlays/UserButton.cs:
```diff
@@ -0,0 +1,150 @@
+            }
+        }
+
+        public Action Action;
```
For the eventual `OnProfile` action

YeahImRose committed:
Various tweaks

YeahImRose committed:
Removed unneeded tooltip container

YeahImRose committed:
Update osu.Desktop.csproj

YeahImRose committed:
Fix for removal of tooltip container

YeahImRose committed:
Working avatar implementation with users

YeahImRose committed:
Merge remote-tracking branch 'refs/remotes/ppy/master'

YeahImRose committed:
Merge branch 'master' of https://github.com/Lunar-Dust/osu-fork

YeahImRose committed:
Update Button.cs

YeahImRose committed:
Merge branch 'master' of https://github.com/Lunar-Dust/osu-fork
fuck merges

YeahImRose committed:
Update APIAccess.cs

Lunar-Dust commented:
Note: This does compile and work properly on my local machine, but refuses to compile on the AppVeyor

peppy commented:
so check the reasons and fix them?

autismpp commented:
You might want to fix the `osu-framework` conflict which will fix the missing `CornerRadius` error.

```
Graphics\Avatar.cs(39,13): error CS0103: The name 'CornerRadius' does not exist in the current context [C:\projects\osu\osu.Game\osu.Game.csproj]
Beatmaps\Timing\SampleChange.cs(15,27): warning CS0649: Field 'SampleChange.Sample' is never assigned to, and will always have its default value null [C:\projects\osu\osu.Game\osu.Game.csproj]
Overlays\UserButton.cs(39,23): warning CS0649: Field 'UserButton.Action' is never assigned to, and will always have its default value null [C:\projects\osu\osu.Game\osu.Game.csproj]
Beatmaps\Timing\TimingChange.cs(14,23): warning CS0649: Field 'TimingChange.BeatLength' is never assigned to, and will always have its default value 0 [C:\projects\osu\osu.Game\osu.Game.csproj]
```

YeahImRose committed:
Updated/Optimized avatar integration

YeahImRose committed:
Allow merging

YeahImRose committed:
Re-fix button

YeahImRose committed:
Improve mergability of PR

This will likely make the PR unable to be compiled on AppVeyor until it is merged

YeahImRose committed:
Undo local changes

YeahImRose committed:
Flip position of username and user avatar

It now looks more like the suggested design

YeahImRose committed:
Further mergability improvements

Lunar-Dust commented:
This compiles properly on AppVeyor, but it will not attempt to build it due to the changes that had to be made to osu.Game.csproj

If testing is required before merging, I recommend doing it locally on a fresh clone.

YeahImRose committed:
Removed redundant check and added comments

YeahImRose committed:
Add comments and some various clean ups

YeahImRose committed:
Make UpdateAvatar asynchronous

YeahImRose committed:
Add comments and slight cleanups

YeahImRose committed:
Update osu-framework version(hopefully)

YeahImRose committed:
Hopefully updating everything won't break everything

YeahImRose committed:
Merge remote-tracking branch 'refs/remotes/ppy/master'

# Conflicts:
#	osu.Game/Overlays/Toolbar.cs
#	osu.Game/osu.Game.csproj

YeahImRose committed:
Merge remote-tracking branch 'refs/remotes/ppy/master'

YeahImRose committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	osu.Game/osu.Game.csproj

YeahImRose committed:
Issue with includes

YeahImRose committed:
Add avatar-guest

YeahImRose committed:
Merge branch 'master' of https://github.com/Lunar-Dust/osu-fork
Stuff merge shit

YeahImRose committed:
add avatar-guest

YeahImRose committed:
Add avatar-guest

YeahImRose committed:
Undo accidental goofiness of osu-resources

Lunar-Dust commented:
Closing to squash

peppy commented:
the fuck?
2016年10月21日(金) 7:46 Lunar Dust <notifications@github.com>:

> This compiles properly on AppVeyor, but it will not attempt to build it
> due to the changes that had to be made to osu.Game.csproj
>
> If testing is required before merging, I recommend doing it locally on a
> fresh clone.
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/ppy/osu/pull/92#issuecomment-255248964>, or mute the
> thread
> <https://github.com/notifications/unsubscribe-auth/AALrZ0PWnLjvm-P8HprxGso4MFzY0FY1ks5q1-8zgaJpZM4KZVhp>
> .
>

Lunar-Dust commented:
I had over 130 commits(mostly from updating from the repo) I would've had to squash on this PR, most of them having merge conflicts on multiple files. Because of this, I decided to make a new PR on a new fork, sorry
